## Module 9 - Case study 2

1.Load the kinematics dataset as measured on mobile sensors from the file “run_or_walk.csv”. List out the columns in the dataset.

2.Let the target variable ‘y’ be the activity and assign all the columns after it to ‘x’.

3.Using Scikit-learn fit a Gaussian Naive Bayes model and observe the accuracy. Generate a classification report using scikit learn.

4.Repeat the model once using only the acceleration values as predictors and then using only the gyro values as predictors. Comment on the difference in accuracy between both the models.

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

C:\Users\sreeraj.va\AppData\Local\Continuum\anaconda3\lib\site-packages\statsmodels\tools\_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [3]:
df = pd.read_csv(r'D:\E\Courses\Edureka\Assignments\Dataset\module9\run_or_walk.csv')

In [4]:
print(df.shape)
df.head(2)

(88588, 11)


date                time username  wrist  activity  acceleration_x  \
0  2017-6-30  13:51:15:847724020   viktor      0         0          0.2650   
1  2017-6-30  13:51:16:246945023   viktor      0         0          0.6722   

   acceleration_y  acceleration_z  gyro_x  gyro_y  gyro_z  
0         -0.7814         -0.0076 -0.0590  0.0325 -2.9296  
1         -1.1233         -0.2344 -0.1757  0.0208  0.1269

In [8]:
df['wrist'].unique()

array([0, 1], dtype=int64)

In [17]:
y = df.iloc[:,4:5].values
x = df.iloc[:, 5:].values

In [18]:
from sklearn.model_selection import train_test_split

x_train, x_test, y_train, y_test = train_test_split(x,y, test_size=0.2, random_state=0)

In [19]:
len(x_train), len(x_test), len(y_train), len(y_test)

(70870, 17718, 70870, 17718)

In [20]:
from sklearn.naive_bayes import GaussianNB
classifier_nb = GaussianNB()
classifier_nb.fit(x_train, y_train)

C:\Users\sreeraj.va\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\utils\validation.py:724: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


GaussianNB(priors=None, var_smoothing=1e-09)

In [22]:
y_pred = classifier_nb.predict(x_test)

In [23]:
from sklearn.metrics import confusion_matrix, accuracy_score, classification_report

In [27]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.92      0.99      0.96      8797
           1       0.99      0.92      0.95      8921

    accuracy                           0.96     17718
   macro avg       0.96      0.96      0.96     17718
weighted avg       0.96      0.96      0.96     17718



In [24]:
confusion_matrix(y_test, y_pred)

array([[8729,   68],
       [ 711, 8210]], dtype=int64)

In [25]:
accuracy_score(y_test, y_pred)

0.9560334123490236

###### 95.6% accuracy score in Gaussian Naive Bayes model with  prediction using all independant variables

### model once using only the acceleration values as predictors

In [30]:
x2 = df.iloc[:,5:8].values

In [31]:
x2_train, x2_test, y_train, y_test = train_test_split(x2,y, test_size=0.2, random_state=0)

In [33]:
len(x2_train), len(x2_test), len(y_train), len(y_test)

(70870, 17718, 70870, 17718)

In [34]:
classifier_nb_accl = GaussianNB()
classifier_nb_accl.fit(x2_train, y_train)

C:\Users\sreeraj.va\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\utils\validation.py:724: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


GaussianNB(priors=None, var_smoothing=1e-09)

In [36]:
y_pred_acc = classifier_nb_accl.predict(x2_test)

In [37]:
print(classification_report(y_test, y_pred_acc))

              precision    recall  f1-score   support

           0       0.92      0.99      0.96      8797
           1       0.99      0.92      0.96      8921

    accuracy                           0.96     17718
   macro avg       0.96      0.96      0.96     17718
weighted avg       0.96      0.96      0.96     17718



In [38]:
confusion_matrix(y_test, y_pred_acc)

array([[8750,   47],
       [ 718, 8203]], dtype=int64)

In [39]:
accuracy_score(y_test, y_pred_acc)

0.9568235692516085

###### 95.68% accuracy score in Gaussian Naive Bayes model with  prediction using only accelaration independant variables

### model once using only the gyro values as predictors

In [42]:
x3 = df.iloc[:,8:].values

In [43]:
x3_train, x3_test, y_train, y_test = train_test_split(x3,y, test_size=0.2, random_state=0)

In [44]:
classifier_nb_gyro = GaussianNB()
classifier_nb_gyro.fit(x3_train, y_train)

C:\Users\sreeraj.va\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\utils\validation.py:724: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


GaussianNB(priors=None, var_smoothing=1e-09)

In [46]:
y_pred_gyro = classifier_nb_gyro.predict(x3_test)

In [47]:
print(classification_report(y_test, y_pred_gyro))

              precision    recall  f1-score   support

           0       0.62      0.74      0.68      8797
           1       0.68      0.55      0.61      8921

    accuracy                           0.65     17718
   macro avg       0.65      0.65      0.64     17718
weighted avg       0.65      0.65      0.64     17718



In [48]:
confusion_matrix(y_test, y_pred_gyro)

array([[6551, 2246],
       [4038, 4883]], dtype=int64)

In [49]:
accuracy_score(y_test, y_pred_gyro)

0.6453324302968733

###### only 64.5% accuracy score in Gaussian Naive Bayes model with  prediction using only gyro independant variables

###### Which means model only with acceleration variables as independant variables is the best. Model with all independant variables is also good.